extensions/standard-processors/processors/TailFile.cpp

```diff
@@ -335,170 +563,210 @@ bool TailFile::storeState(const std::shared_ptr<core::ProcessContext>& context)
-
-        for (auto ffr : flowFiles) {
-          logger_->log_info("TailFile %s for %u bytes", state.first, ffr->getSize());
-          std::string logName = baseName + "." + std::to_string(state.second.currentTailFilePosition_) + "-" + std::to_string(state.second.currentTailFilePosition_ + ffr->getSize()) + "." + extension;
-          ffr->updateKeyedAttribute(PATH, fileLocation);
-          ffr->addKeyedAttribute(ABSOLUTE_PATH, fullPath);
-          ffr->updateKeyedAttribute(FILENAME, logName);
-          session->transfer(ffr, Success);
-          state.second.currentTailFilePosition_ += ffr->getSize() + 1;
-          storeState(context);
-        }
+  std::size_t last_dot_position = fileName.find_last_of('.');
+  std::string baseName = fileName.substr(0, last_dot_position);
+  std::string extension = fileName.substr(last_dot_position + 1);
+
+  if (!delimiter_.empty()) {
+    char delim = delimiter_[0];
+    logger_->log_trace("Looking for delimiter 0x%X", delim);
+
+    std::size_t num_flow_files = 0;
+    FileReaderCallback file_reader{full_file_name, state.position_, delim, state.checksum_};
+    TailState state_copy{state};
+
+    while (file_reader.hasMoreToRead()) {
+      auto flow_file = std::static_pointer_cast<FlowFileRecord>(session->create());
+      session->write(flow_file, &file_reader);
+
+      if (file_reader.useLatestFlowFile()) {
+        updateFlowFileAttributes(full_file_name, state_copy, fileName, baseName, extension, flow_file);
+        session->transfer(flow_file, Success);
+        updateStateAttributes(state_copy, flow_file->getSize(), file_reader.checksum());
+
+        ++num_flow_files;
 
       } else {
-        std::shared_ptr<FlowFileRecord> flowFile = std::static_pointer_cast<FlowFileRecord>(session->create());
-        if (flowFile) {
-          flowFile->updateKeyedAttribute(PATH, fileLocation);
-          flowFile->addKeyedAttribute(ABSOLUTE_PATH, fullPath);
-          session->import(fullPath, flowFile, true, state.second.currentTailFilePosition_);
-          session->transfer(flowFile, Success);
-          logger_->log_info("TailFile %s for %llu bytes", state.first, flowFile->getSize());
-          std::string logName = baseName + "." + std::to_string(state.second.currentTailFilePosition_) + "-" + std::to_string(state.second.currentTailFilePosition_ + flowFile->getSize()) + "."
-              + extension;
-          flowFile->updateKeyedAttribute(FILENAME, logName);
-          state.second.currentTailFilePosition_ += flowFile->getSize();
-          storeState(context);
-        }
+        session->remove(flow_file);
       }
-      state.second.currentTailFileModificationTime_ = ((uint64_t) (statbuf.st_mtime) * 1000);
-    } else {
-      logger_->log_warn("Unable to stat file %s", fullPath);
     }
+
+    state = state_copy;
+    storeState(context);
+
+    logger_->log_info("%u flowfiles were received from TailFile input", num_flow_files);
```

Comment:
Please use `zu` format specifier for `size_t`

Reply:
done